src/main/java/org/xbill/DNS/config/AndroidResolverConfigProvider.java

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: BSD-2-Clause
+package org.xbill.DNS.config;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.xbill.DNS.ResolverConfig;
+import org.xbill.DNS.SimpleResolver;
+
+/**
+ * Resolver config provider for Android. Contrary to all other providers, this provider needs a
+ * context to operate on which must be set by calling {@link #setContext(Object)}.
+ *
+ * <p>If you are developing for Android, consider implementing your own {@link
+ * ResolverConfigProvider} that listens to network callbacks and properly refreshes on link changes.
```

Comment:
Can you define "link changes" further?  Does that just mean network changes?

Reply:
Yes, I'm not developing for Android, so I can only assume things from reading the docs. `ConnectivityManager.registerNetworkCallback` and `registerDefaultNetworkCallback` seem to be interesting APIs.